### PR TITLE
feat(*): Bumps our wasmcloud wit to 1.0

### PR DESCRIPTION
**Pull request by thomastaylor312**
This bumps our main `wasmcloud:bus` package and the package in the `runtime` crate to 1.0.0 in preparation for release. Also removes the use of the wasmcloud interface in tests where it wasn't needed

### CODE REVIEW

**Comment by thomastaylor312** [comment]
Please note that this is stacked on top of #1809 to avoid merge conflicts. I will rebase once the other is merged

File: wit/bus.wit
Comment (brooksmtownsend) [inline wit/bus.wit:1]: I see a pretty significant TODO in this WIT, do we want to make sure this interface is ready for stability?

What I think of as the (pseudocode) usage of this interface is:
```rust
wasi::keyvalue::set("foo", "bar"); // sets a key in whatever is linked on default
wasmcloud::bus::lattice::set_link_name("secret", "wasi:keyvalue/set");
wasi::keyvalue::set("foo", "p4ssw0rd"); // sets a key in whatever is linked on secret
```

If we removed the interfaces, then we could do the same as above with a little less code at the cost of setting the link name for every interface call, which makes me think the above is still necessary.

Comment (brooksmtownsend) [reply wit/bus.wit:1]: A conversation for another PR is whether or not this implementation in the host works with different component instances (cc @vados-cosmonic, since we spoke about this)

File: wit/README.md
Comment (brooksmtownsend) [inline wit/README.md:1]: I think it would be great to include a small README here just noting that this WIT interface is used for the advanced use case of multiple links on the same interface (you could leave it at that.) Optional pointer to where the full list of WIT things we support (runtime?)

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
Version bumps look great to me. Requesting changes since I think we should figure out these issues before we call the wits 1.0, but I don't think we have to hold this specific PR back as long as we're OK with it changing in `main`

File: wit/README.md
Comment (thomastaylor312) [reply wit/README.md:1]: I’d rather document it in the interface itself since docs can be generated from that

File: wit/bus.wit
Comment (thomastaylor312) [reply wit/bus.wit:1]: We should actually figure that out here before we bump to 1.0

Comment (thomastaylor312) [reply wit/bus.wit:1]: So I think we should keep interfaces just so it can be more focused. If you have one interface that only uses a custom link name and the others are default, it would really suck to have to unset the link name with every call. I think we can make some nice wrappers for this that could go into `wasmcloud_actor` eventually

File: wit/README.md
Comment (thomastaylor312) [reply wit/README.md:1]: Documentation added!

**Comment by thomastaylor312** [comment]
This will be ready to go once #1815 is merged. I'll rebase on that just to be sure before merging this

**Review by brooksmtownsend — APPROVED** [review]

**Pull request merged**
